src/qcodes/dataset/exporters/export_to_pandas.py

```diff
@@ -12,17 +12,24 @@
     import pandas as pd
 
     from qcodes.dataset.data_set_protocol import ParameterData
+    from qcodes.dataset.descriptions.dependencies import InterDependencies_
 
 
-def load_to_dataframe_dict(datadict: ParameterData) -> dict[str, pd.DataFrame]:
+def load_to_dataframe_dict(
+    datadict: ParameterData, interdeps: InterDependencies_
+) -> dict[str, pd.DataFrame]:
     dfs = {}
     for name, subdict in datadict.items():
-        index = _generate_pandas_index(subdict)
+        index = _generate_pandas_index(
+            subdict, interdeps=interdeps, top_level_param_name=name
+        )
         dfs[name] = _data_to_dataframe(subdict, index)
     return dfs
 
 
-def load_to_concatenated_dataframe(datadict: ParameterData) -> pd.DataFrame:
+def load_to_concatenated_dataframe(
+    datadict: ParameterData, interdeps: InterDependencies_
+) -> pd.DataFrame:
```

Comment:
Changing these methods is technically an API break. I think that is acceptable as this is not documented